mbuild/compound.py

```diff
@@ -1347,7 +1352,24 @@ def get_boundingbox(self):
         for i, dim in enumerate(has_dimension):
             if not dim:
                 vecs[i][i] = 1.0
-        return Box.from_vectors(vectors=np.asarray([vecs]).reshape(3, 3))
+        try:
```

Comment:
So it seems to me that this is a little confusing. Here is the case that I think could be a little strange for users:

I for whatever reason need a padded boundingbox. Maybe it's for a writer, maybe it's for some packmol usage, or maybe it's so I can increase my initial volume slightly. I get to the get_boundingbox method, and see the option to pad_box. I could in that case set pad_box to `True` and think that I added the 1 nm padding, but I really am not because no MBuildError is raised with the `Compound` I'm passing.

To summarize, I think a more flexible and general addition to this function could be to set the default for `pad_box` to None, but if you pass a `float` type to `pad_box`, then you could pad it by however much you like. I could even see a possibility where you could pass a list of [float, float, float] and have each of those paddings applied to your three independent box vectors. In this case, you could pad whichever dimension is being identified as a co-linear vector, and just pad that dimension.

This seems like a different solution, let me know if you like this or think that would be a beneficial change.

Reply:
Great suggestion! I am on board with that change as well. 

The behavior I think would be:
* Have `pad_box` as `None` by default
    *  if co-linear vectors, error out
* allow `pad_box` to be a single float
    * Pad all box edges by float value
* allow `pad_box` to be a list of 3 floats
    *  Pad a, b, c box edges by their respective value in the list